modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1120,6 +1120,9 @@ def to_pandas_remote(df, partition_shape, *dfs):
                 (df,) + dfs, partition_shape, called_from_remote=True
             )
 
+        if partitions.size == 0:
```

Comment:
@arunjose696 please add also a test

Reply:
Wrote the test, the fix is slightly complicated as the broadcasted right dataframes being empty could not be reconstructed as the partitions would be empty (and a simple reconstruction of empty datatframe does not works as  the column information is not available).

 I assume I would need to modify broadcast_apply_full_axis logic as other modin operations eg. df.compare(df2) which use broadcast_apply_full_axis would also fail if the second dataframe is empty.

Reply:
In the case of empty partitions, could we also send data about the indexes that we store at the dataframe level?

Reply:
I have tried an approach with creating a new partition (with indexes and columns from dataframe level)in broadcast_apply_full axis if the dataframe does not have partitions, this seems to work for the test case.